src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1196,6 +1196,14 @@ private Task<HttpResponseMessage> SendRequest(bool async, HttpContent? content =
                 if (IsWellKnownContentHeader(headerName))
                 {
                     _sendRequestMessage.Content ??= new ByteArrayContent(Array.Empty<byte>());
+
+                    if (headerName == HttpKnownHeaderNames.ContentLength)
```

Comment:
ignore case as in `IsWellKnownContentHeader`?
```suggestion
                    if (string.Equals(headerName, HttpKnownHeaderNames.ContentLength, StringComparison.OrdinalIgnoreCase))
```